pages/tutorials/3.0/window/events.md

```diff
@@ -7,259 +7,329 @@
+The `sf::Event::Closed` event is triggered when the user wants to close the window through any of the possible methods the window manager provides ("close" button, keyboard shortcut, etc.).
+This event only represents a close request.
+The window is not yet closed when the event is received.
 
-Typical code will just call `window.close()` in reaction to this event, to actually close the window. However, you may also want to do something else first, like saving the current application state or asking the user what to do. If you don't do anything, the window remains open.
+Typically, one will just call `window.close()` in reaction to this event to actually close the window.
+However, you may also want to do something else first like saving the current application state or asking the user what to do.
+If you don't do anything, the window remains open.
 
-There's no member associated with this event in the [`sf::Event`](https://www.sfml-dev.org/documentation/3.0.0/classsf_1_1Event.php "sf::Event documentation") union.
+There's no data associated with this event.
 
 ```cpp
-if (event.type == sf::Event::Closed)
+if (event->is<sf::Event::Closed>())
     window.close();
 ```
 
 ## The Resized event
 
 The `sf::Event::Resized` event is triggered when the window is resized, either through user action or programmatically by calling `window.setSize`.
-
 You can use this event to adjust the rendering settings: the viewport if you use OpenGL directly, or the current view if you use sfml-graphics.
 
-The member associated with this event is `event.size`, it contains the new size of the window.
+There's no data associated with this event.
 
 ```cpp
-if (event.type == sf::Event::Resized)
+if (event->is<sf::Event::Resized>())
 {
     std::cout << "new width: " << event.size.width << std::endl;
     std::cout << "new height: " << event.size.height << std::endl;
 }
 ```
 
-## The LostFocus and GainedFocus events
-
-The `sf::Event::LostFocus` and `sf::Event::GainedFocus` events are triggered when the window loses/gains focus, which happens when the user switches the currently active window. When the window is out of focus, it doesn't receive keyboard events.
+## The FocusLost and FocusGained events
 
+The `sf::Event::FocusLost` and `sf::Event::FocusGained` events are triggered when the window loses/gains focus, which happens when the user switches the currently active window.
+When the window is out of focus, it doesn't receive keyboard events.
 This event can be used e.g. if you want to pause your game when the window is inactive.
 
-There's no member associated with these events in the [`sf::Event`](https://www.sfml-dev.org/documentation/3.0.0/classsf_1_1Event.php "sf::Event documentation") union.
+There's no data associated with this event.
```

Comment:
```suggestion
There's no event data associated with this event.
```

Reply:
Are you sure? It feels awkward to repeat the word `event` twice.

Reply:
Data is such a generic word, that I struggle with it, when it's not more explicitly defined.
It could be some random data attached to the event, or it could be data specifically related to the given event.
That's why I feel "event data" makes it clearer what kind of data we're talking about here.